pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -466,3 +561,38 @@ func fromRecipeConfigEnvDatamodel(config datamodel.RecipeConfigProperties) map[s
 
 	return env
 }
+
+func toRecipeConfigEnvSecretsDatamodel(config *RecipeConfigProperties) map[string]datamodel.SecretReference {
+	if config.EnvSecrets == nil {
+		return nil
+	}
+
+	envSecrets := map[string]datamodel.SecretReference{}
+	for k, v := range config.EnvSecrets {
+		envSecrets[k] = datamodel.SecretReference{
+			Source: to.String(v.Source),
+			Key:    to.String(v.Key),
+		}
+	}
+
+	return envSecrets
+}
+
+func fromRecipeConfigEnvSecretsDatamodel(config datamodel.RecipeConfigProperties) map[string]*SecretReference {
+	var envSecrets map[string]*SecretReference
+
+	for k, v := range config.EnvSecrets {
+		convertedEnvSecret := &SecretReference{
+			Source: to.Ptr(v.Source),
+			Key:    to.Ptr(v.Key),
+		}
+
+		if envSecrets == nil {
```

Comment:
Why is this check needed?

Reply:
this function has been refactored. the check is needed to ensure we instantiate only when needed